ngraph/test/backend/unary_test.hpp

```diff
@@ -0,0 +1,211 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include <algorithm>
+#include <cinttypes>
+#include <cmath>
+#include <cstdlib>
+#include <memory>
+#include <random>
+#include <string>
+#include <tuple>
+
+#include "gtest/gtest.h"
+#include "ngraph/function.hpp"
+#include "ngraph/ngraph.hpp"
+#include "util/engine/test_engines.hpp"
+#include "util/test_case.hpp"
+#include "util/test_control.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+namespace
+{
+    // Tolerance can be specified in two forms (number of bits in mantissa or float-point
+    // threshold), corresponding to two different comparing methods implemented in TestCase
+    // classes. Following class acts as an encapsulation on these two options without introducing
+    // too many overloadings on the function requiring such argument.
+    struct ArgTolerance
+    {
+        const int tolerance_bits;
+        const float tolerance_fp;
+        bool using_bits;
+        ArgTolerance()
+            : tolerance_bits(-1)
+            , tolerance_fp(0)
+            , using_bits(true)
+        {
+        }
+        ArgTolerance(const int tolerance_bits)
+            : tolerance_bits(tolerance_bits)
+            , tolerance_fp(0)
+            , using_bits(true)
+        {
+        }
+        ArgTolerance(const float tolerance_fp)
+            : tolerance_bits(-1)
+            , tolerance_fp(tolerance_fp)
+            , using_bits(false)
+        {
+        }
+        ArgTolerance(const double tolerance_fp)
+            : tolerance_bits(-1)
+            , tolerance_fp(tolerance_fp)
+            , using_bits(false)
+        {
+        }
+    };
```

Comment:
```suggestion
    template<typename V, typename Tag>
    struct ArgumentTolerance 
    {
        ArgumentTolerance() = default;
        ArgumentTolerance(V v) : in_use{ture},value{v}
        bool in_use{false};
        V value{};
     };
     using BitTolerance = ArgumentTolerance<int, struct BitTag>;
     using FloatTolerance = ArgumentTolerance<float, struct FloatTag>;
    struct Tolerance
    {
        BitTolerance bit;
        FloatTolerance floatingpoint;
    };
```

Reply:
Good idea, it's more clear and flexible. One small change is, to allow caller pass an int or float value as tolerant, I would add back those constructors for class Tolerance.